tempesta_fw/http_sess.c

```diff
@@ -361,7 +359,7 @@ tfw_http_sticky_notfound(TfwHttpReq *req)
 
 	/* Create Tempesta sticky cookie and store it */
 	if (tfw_http_sticky_calc(req, &sv) != 0)
-		return tfw_http_send_502(req);
+		return -1;
```

Comment:
Good catch and the fix eliminates the problem.

However now `tfw_http_req_cache_cb()` sends 500 error code which is generic code since the function doesn't know exact reason for the error. Meantime previously we sent 502nd error code. 502 error code seems doubtful in this situation since we failed on sticky cookie calculation and no upstream communication was involved, so I think we shouldn't return "Bad gateway code". Now behaviour seems correct.

The design problem is that there are different subsystems (Web-cache, load balancing schedulers etc.) which can return with errors and we have to send correct error codes to clients. One way to keep design correct is to return negative HTTP error codes (i.e. `-500` in this case) such that upper layer can send correct HTTP error messages. The second way (seems what we have now) is to just make decision which error code to send at `http.c` layer depending on which subsystem fails.